core/trino-spi/src/main/java/io/trino/spi/type/RowType.java

```diff
@@ -426,6 +426,15 @@ private List<OperatorMethodHandle> getReadValueOperatorMethodHandles(TypeOperato
                 new OperatorMethodHandle(WRITE_FLAT_CONVENTION, writeFlat));
     }
 
+    public static SqlRow getSqlRowFromFieldBuilders(BlockBuilder[] fieldBuilders)
```

Comment:
1. Maybe move this to `RowBlockBuilder` so that the pair of utility methods are together
2. Maybe rename to `createSqlRowFromFieldBuildersForSingleRow`